### PR TITLE
build: dockerfile

### PR DESCRIPTION
**Pull request by WatcharitNew**
# Overview
- Add `dockerfile` and `docker-compose`
- Manage `env`

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+WORKDIR /usr/src/app
+
+# Copy the built application from the previous stage
+COPY --from=build /usr/src/app/.next ./.next
+COPY --from=build /usr/src/app/public ./public
+COPY --from=build /usr/src/app/node_modules ./node_modules
+
+RUN ln -sf /usr/share/zoneinfo/Asia/Bangkok /etc/localtime
+
+CMD ["npx", "next", "start", "-p", "3000"]
```
Comment (yutna) [inline Dockerfile:34]: Not sure, but can we simplify it by just using `yarn start`?

File: src/config/env.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+const branch = process.env.NEXT_PUBLIC_BRANCH;
+
+interface Env {
+  BASE_STORYBOOK_URL: string;
+  NEXT_PUBLIC_BASE_URL: string;
+  NEXT_PUBLIC_BASE_STORYBOOK_URL: string;
+  IS_PROD: boolean;
+}
+
+const envObj: Env = {
```
Comment (yutna) [inline src/config/env.ts:10]: const ENV: Env = { .. };

File: src/config/env.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+const branch = process.env.NEXT_PUBLIC_BRANCH;
+
+interface Env {
```
Comment (yutna) [inline src/config/env.ts:3]: This should be moved to the `types` folder (`types/env.ts`) and imported into this file using `import type`.

File: next.config.mjs [hunk 4]
```diff
@@ -19,6 +19,11 @@ const nextConfig = {
     // https://nextjs.org/docs/app/building-your-application/configuring/typescript#statically-typed-links
     typedRoutes: true,
   },
+  env: {
+    NEXT_PUBLIC_BASE_URL: process.env.NEXT_PUBLIC_BASE_URL,
+    NEXT_PUBLIC_BASE_STORYBOOK_URL: process.env.NEXT_PUBLIC_BASE_STORYBOOK_URL,
+    NEXT_PUBLIC_BRANCH: process.env.NEXT_PUBLIC_BRANCH,
+  },
```
Comment (yutna) [inline next.config.mjs:26]: Not sure but, Is this necessary? Next.js actually uses `NEXT_PUBLIC` by default. 🤔

**Review by yutna — CHANGES_REQUESTED** [review]

File: next.config.mjs (hunk 4 above)
Comment (WatcharitNew) [reply next.config.mjs:26]: Thank you krub. I will remove then 🙏

**Review by yutna — APPROVED** [review]
LGTM ❤️

**Pull request merged**